scrapy/crawler.py

```diff
@@ -72,9 +72,11 @@ def crawl(self, *args, **kwargs):
             start_requests = iter(self.spider.start_requests())
             yield self.engine.open_spider(self.spider, start_requests)
             yield defer.maybeDeferred(self.engine.start)
-        except Exception:
+        except Exception as e:
             self.crawling = False
-            raise
+            if self.engine is not None:
+                yield self.engine.close()
+            raise e
```

Comment:
By changing `raise` to `raise e` you're loosing the original traceback - is it intentional?